services/graph/pkg/service/v0/driveitems.go

```diff
@@ -731,34 +741,71 @@ func (g Graph) DeletePermission(w http.ResponseWriter, r *http.Request) {
 
 }
 
-func (g Graph) getPermissionByID(ctx context.Context, permissionID string) (*libregraph.Permission, *storageprovider.ResourceId, error) {
-	share, err := g.getCS3UserShareByID(ctx, permissionID)
+func (g Graph) getPermissionByID(ctx context.Context, permissionID string, itemID *storageprovider.ResourceId) (*libregraph.Permission, *storageprovider.ResourceId, error) {
+	publicShare, err := g.getCS3PublicShareByID(ctx, permissionID)
 	if err == nil {
-		permission, err := g.cs3UserShareToPermission(ctx, share)
+		permission, err := g.libreGraphPermissionFromCS3PublicShare(publicShare)
 		if err != nil {
 			return nil, nil, err
 		}
-		return permission, share.GetResourceId(), nil
+		return permission, publicShare.GetResourceId(), nil
+	}
+
+	// The id is not referencing a public share, if the itemID is referencing
+	// a spaceroot this is a space permission. Handle that next
+	if IsSpaceRoot(itemID) {
+		gatewayClient, err := g.gatewaySelector.Next()
+		if err != nil {
+			g.logger.Debug().Err(err).Msg("selecting gatewaySelector failed")
+			return nil, nil, err
+		}
+		// get space id
+		statResponse, err := gatewayClient.Stat(ctx, &storageprovider.StatRequest{Ref: &storageprovider.Reference{ResourceId: itemID}})
+		if errCode := errorcode.FromStat(statResponse, err); errCode != nil {
+			g.logger.Warn().Err(errCode).Interface("stat.res", statResponse).Msg("stat failed")
+			return nil, nil, errCode
+		}
```

Comment:
You can use `utils.GetResourceByID` if you don't want to hassle with stat calls. (not critical ofc)

Reply:
Hm, nice